### PR TITLE
ci: update deps

### PR DESCRIPTION
**Pull request by artemrys**
Update some of the CI dependencies, remove `review_secrets` job and remove `dependabot.yml` config as well.

### CODE REVIEW

File: .github/workflows/build-test-release.yml [hunk 1]
```diff
@@ -49,47 +49,32 @@ jobs:
...
-    if: github.actor != 'dependabot[bot]'
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - id: semgrep
         uses: semgrep/semgrep-action@v1
         with:
           publishToken: ${{ secrets.SEMGREP_PUBLISH_TOKEN }}
 
-  review_secrets:
```
Comment (mkolasinski-splunk) [inline .github/workflows/build-test-release.yml:75]: What is the reason for removing trufflehog job?

Comment (mkolasinski-splunk) [reply .github/workflows/build-test-release.yml:75]: I mean - this one is old, but imo we should rather replace it with https://github.com/marketplace/actions/trufflehog-oss

Comment (artemrys) [reply .github/workflows/build-test-release.yml:75]: I don't find it useful and it does not seem to be a requirement to have it.

**Review by mkolasinski-splunk — APPROVED** [review]

**Pull request merged**